lib/android/app/src/main/java/com/reactnativenavigation/viewcontrollers/stack/StackPresenter.java

```diff
@@ -420,6 +420,7 @@ private void mergeLeftButton(TopBarOptions options, TopBarButtons buttons, View
         List<ButtonController> toRemove = difference(currentLeftButtons, toMerge, ButtonController::areButtonsEqual);
         forEach(toRemove, ButtonController::destroy);
         if (!CollectionUtils.equals(currentLeftButtons, toMerge)) {
+            topBar.clearBackButton();
```

Comment:
Wouldn't it be better to invoke `clearBackButton` in `topBarController.mergeLeftButtons()`? This way it's the topBarConroller's responsibility to ensure it's in correct state.